src/libAtomVM/term.h

```diff
@@ -1211,6 +1279,114 @@ static inline uint64_t term_to_ref_ticks(term rt)
     #endif
 }
 
+/**
+ * @brief Get a pid term from node, process_id, serial and creation
+ *
+ * @param node name of the node (atom)
+ * @param process_id process id on that node
+ * @param serial serial of process id on that node
+ * @param creation creation of that node
+ * @param heap the heap to allocate memory in
+ * @return an external heap term created using given parameters.
+ */
+static inline term term_from_external_process_id(term node, uint32_t process_id, uint32_t serial, uint32_t creation, Heap *heap)
+{
+    term *boxed_value = memory_heap_alloc(heap, EXTERNAL_PID_SIZE);
+    int atom_index = term_to_atom_index(node);
+    boxed_value[0] = (atom_index << 6) | TERM_BOXED_EXTERNAL_PID;
+
+    #if TERM_BYTES == 8
+        boxed_value[1] = (term) (((uint64_t) process_id) << 32 | creation);
+        boxed_value[2] = (term) serial;
+
+    #elif TERM_BYTES == 4
+        boxed_value[1] = (term) creation;
+        boxed_value[2] = (term) process_id;
+        boxed_value[3] = (term) serial;
+
+    #else
+        #error "terms must be either 32 or 64 bit wide"
+    #endif
+
+    return ((term) boxed_value) | TERM_BOXED_VALUE_TAG;
+}
+
+/**
+ * @brief Get the name of a node for a given external thing
+ *
+ * @param term external term
+ * @return the name of the node
+ */
+static inline term term_to_external_node(term t)
+{
+    TERM_DEBUG_ASSERT(term_is_external(t));
+
+    const term *boxed_value = term_to_const_term_ptr(t);
+
+    return TERM_FROM_ATOM_INDEX(boxed_value[0] >> 6);
+}
+
+/**
+ * @brief Get the creation for a given external thing
+ *
+ * @param term external term
+ * @return the serial of the external pid
+ */
+static inline uint32_t term_to_external_node_creation(term t)
```

Comment:
I think the name should be changed to something else.
I would use the "to" and "from" when there is some kind of straightforward conversion that can be done back and forth (as for integers).
Here I would use `term_get_pid_external_node_creation`, and I would do something similar for other functions.